examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py

```diff
@@ -115,6 +116,22 @@ class TaskInfo(BaseModel):
     metadata: Dict[str, Any]
 
 
+class DagRun(BaseModel):
+    dag_id: str
+    note: str
+    state: str
+    airflow_run_id: str
+    start_date: datetime.datetime
+    end_date: datetime.datetime
+    run_type: str
+    airflow_run_config: Dict[str, Any]
+    raw_metadata: Dict[str, Any]
+
+    @property
+    def success(self) -> bool:
+        return self.state == "success"
```

Comment:
when are using `BaseModel` versus `record` versus plain ole python?

Reply:
This was mostly me just being inconsistent. Some stuff was originally basemodel for yaml parsing purposes, but I think most of these should just be record.

I changed the relevant classes to record.